app/views/AuthenticationWebView.js

```diff
@@ -58,37 +62,58 @@ class OAuthView extends React.PureComponent {
 		this.setState({ logging: true });
 
 		try {
-			await RocketChat.loginOAuth(params);
+			await RocketChat.loginOAuthOrSso(params);
 		} catch (e) {
 			console.warn(e);
 		}
 		this.setState({ logging: false });
 		this.dismiss();
 	}
 
+	onNavigationStateChange = (webViewState) => {
+		const url = decodeURIComponent(webViewState.url);
+		if (this.authType === 'saml' || this.authType === 'cas') {
+			const { navigation } = this.props;
+			const ssoToken = navigation.getParam('ssoToken');
+			if (url.includes('ticket') || url.includes('validate')) {
+				let payload;
+				if (this.authType === 'saml') {
+					payload = `{ "saml": true, "credentialToken": "${ ssoToken }" }`;
```

Comment:
Why you not use a simple object?

Reply:
Is it more efficient @djorkaeffalexandre?

Reply:
@filipedelimabrito I think `object` is more clean that `string + json.parse`, but is only a suggestion hahaha.

Reply:
You can use:
```
const credentialToken = { credentialToken: ssoToken };
if (this.authType === ‘saml’) {
  payload = { ...credentialToken, saml: true };
} else {
  payload = { cas: credentialToken };
}
```

Reply:
Thanks @djorkaeffalexandre! It's more clean now!